zerver/models.py

```diff
@@ -814,6 +814,21 @@ def major_tos_version(self):
         else:
             return -1
 
+class Team(models.Model):
+    team_name = models.CharField(max_length=100)
+    members = models.ManyToManyField(UserProfile, through='TeamMembership')
+    realm = models.ForeignKey(Realm)
+
+    class Meta:
+        unique_together = (('team_name', 'realm'),)
+
+class TeamMembership(models.Model):
+    team = models.ForeignKey(Team)
+    member = models.ForeignKey(UserProfile)
```

Comment:
Let's make this `user_profile`; that's our convention in basically all models.

Reply:
Sure.